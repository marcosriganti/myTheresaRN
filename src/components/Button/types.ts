import {PressableProps} from "react-native";
type ButtonType = 'primary' | 'secondary' | 'tertiary';
interface ButtonProps extends PressableProps {
    type?: ButtonType;
}

export type {
    ButtonProps,
    ButtonType,
};
