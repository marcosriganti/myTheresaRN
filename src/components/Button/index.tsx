import React from 'react';
import {
    Text,
    Pressable,
} from 'react-native';
import styled from 'styled-components';
import {ButtonType, ButtonProps} from './types';

const StyledButton = styled(Pressable) <{type?: ButtonType;}>`
    background-color: #1d2bb2;
    border-radius: 8px;
    padding: 8px;
`;
const StyledButtonText = styled(Text)`
    color: #FFFFFF;
    text-align: center;
`;

const Button = ({children, type = 'primary', ...rest}: ButtonProps & {children: React.ReactNode | string;}) => {
    return <StyledButton type={type} {...rest}><StyledButtonText>{children}</StyledButtonText></StyledButton>;
};


export default Button;
