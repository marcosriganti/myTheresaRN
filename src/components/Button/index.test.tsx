import 'react-native';
import React from 'react';
import Button from './index';

import {render} from '@testing-library/react-native';
import {fireEvent} from '@testing-library/react-native';


describe('Button', () => {
    it('renders correctly', () => {
        const {toJSON} = render(<Button>Test Button</Button>);
        expect(toJSON()).toMatchSnapshot();
    });

    it('displays the correct text', () => {
        const {getByText} = render(<Button>Test Button</Button>);
        expect(getByText('Test Button')).toBeTruthy();
    });

    it('handles press events', () => {
        const mockOnPress = jest.fn();
        const {getByText} = render(<Button onPress={mockOnPress}>Test Button</Button>);
        fireEvent.press(getByText('Test Button'));
        expect(mockOnPress).toHaveBeenCalled();
    });
});