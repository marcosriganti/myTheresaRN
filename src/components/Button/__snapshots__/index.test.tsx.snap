// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  className="sc-blHHSb cxpjSJ"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  type="primary"
>
  <Text
    className="sc-gtLWhw fPSmTe"
  >
    Test Button
  </Text>
</View>
`;
